challenge_01/script.js

```diff
@@ -0,0 +1,60 @@
+const contactForm = document.querySelector("#contact-form");
+const firstNameInput = document.querySelector("#firstname-input");
+const lastNameInput = document.querySelector("#lastname-input");
+const emailInput = document.querySelector("#email-input");
+const radioBtnGE = document.querySelector("#radio-btn-GE");
+const radioBtnSR = document.querySelector("#radio-btn-SR");
+const messageTextarea = document.querySelector("#message-textarea");
+const consentCheckBox = document.querySelector("#consent-checkbox");
+
+contactForm.addEventListener('submit',(event) => {
+
+    // disabling the default submit and refresh
+    event.preventDefault();
+
+    //checking for wrong inputs
+    if (!firstNameInput.value){
+        alert("please enter first name");
+        return;
+    }
+    if (firstNameInput.value.length < 4 || firstNameInput.value.length > 10){
+        alert("first name length should be between 4 and 10 letters");
+        return;
+    }
+    if (!lastNameInput.value){
+        alert("please enter last name");
+        return;
+    }
+    if (lastNameInput.value.length < 4 || lastNameInput.value.length > 10){
+        alert("last name length should be between 4 and 10 letters");
+        return;
+    }
+    if (!emailInput.value){
+        alert("please enter email");
+        return;
+    }
+    if (!emailInput.value.match(/^[\w.+\-]+@gmail\.com$/)){
+        alert("please enter a valid \"gmail\" address");
+        return;
+    }
+    if (!radioBtnGE.checked && !radioBtnSR.checked){
+        alert("please choose one of the options");
+        return;
+    }
+    if (!messageTextarea.value){
+        alert("please enter a message");
+        return;
+    }
+    if (!consentCheckBox.checked){
+        alert("please check the consent checkbox to continue")
+        return;
```

Comment:
Our challenge is not the real world, I know that, but let's say, there is a new requirement tomorrow which says: 

- for every alert message, perform one console.error("the_very_same_message_you_passed_to_alert")
- don't return, let the user submit the form (for analytical use cases maybe, who knows)

Then you would have to touch 9 different if branches to implement this. To fix this you can extract the common functionality and put it into a function like:
```js
let shouldReturnEarly = false;

function formErrored(message, options) {
  alert(message)
  console.error(message)
  shouldReturn = true;
}

if (whatever) formErrored("name is empty")
// many if branches

if (shouldReturnEarly) return
submit()
```

Reply:
Done.
Also I realized in this scenario my code would alert 2 errors for name and email inputs. For example one to say that input is empty and other one to say that input should be between 4 and 10 letters. So I replaced second `if`s of each input with `else if`s.
 Instead of:
`if (!formElements.firstNameInput.value){
        formErrorHandler("please enter first name");
    }
if (formElements.firstNameInput.value.length < 4 || formElements.firstNameInput.value.length > 10){
        formErrorHandler("first name length should be between 4 and 10 letters");
    }`
Now it's:
`if (!formElements.firstNameInput.value){
        formErrorHandler("please enter first name");
    }
    **else** if (formElements.firstNameInput.value.length < 4 || formElements.firstNameInput.value.length > 10){
        formErrorHandler("first name length should be between 4 and 10 letters");
    }`